platforms/documentation/docs/src/docs/userguide/reference/dsl-apis/public_apis.adoc

```diff
@@ -18,192 +18,68 @@
 
 Gradle provides many public APIs that allow developers to interact with various build system components.
 
-Any class name containing `Internal`, or that is located in a package containing the segment `.internal.` is by definition not part of the public Gradle API and should not be relied upon for stable usage.
-
-The following table summarizes public Gradle APIs and their common use cases.
-This is *not* an exhaustive list:
-
-[cols="1,1,2"]
+Any class or package name containing `Internal`, or located in a package segment `.internal.`, is not part of the public Gradle API and should not be relied upon for stable usage.
+Use of Gradle internal APIs in plugins and build scripts can break builds when either Gradle or plugins change.
+
+The list of public API packages is maintained in the link:https://github.com/gradle/gradle/blob/master/buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/public-api.kt[Gradle public API definition] (excluding any subpackages containing internal).
+It is also available below:
+
+[source,text]
+----
+org/gradle/*
```

Comment:
A couple issues:
- I think these should be listed with `.` instead of `/`.
- I think this is unclear regardless -- these are referred to as "packages" but the globs here actually refer to classes, i.e. `org.gradle.Foo` is public, but `org.gradle.something.Foo` is internal. But that's not clear from the `*` vs `**` alone. Perhaps we can give this particular entry a special note that it doesn't include subpackages?